packages/Chem/src/rdkit-service/rdkit-service-worker-similarity.ts

```diff
@@ -17,43 +18,91 @@ export class RdKitServiceWorkerSimilarity extends RdKitServiceWorkerBase {
     const fps: (Uint8Array | null)[] = [];
     try {
       switch (fingerprintType) {
-      case Fingerprint.Pattern:
-        for (let i = 0; i < this._rdKitMols.length; ++i) {
-          try {
-            if (!this._rdKitMols[i])
+        case Fingerprint.Pattern:
+          for (let i = 0; i < this._rdKitMols.length; ++i) {
+            try {
+              if (!this._rdKitMols[i])
+                fps.push(null);
+              else
+                fps.push(this._rdKitMols[i]!.get_pattern_fp_as_uint8array());
+            } catch {
               fps.push(null);
-            else
-              fps.push(this._rdKitMols[i]!.get_pattern_fp_as_uint8array());
-          } catch {
-            fps.push(null);
-            continue;
+              continue;
+            }
           }
-        }
-        break;
-      case Fingerprint.Morgan:
-        for (let i = 0; i < this._rdKitMols.length; ++i) {
-          try{
-            if (!this._rdKitMols[i] || this._rdKitMols[i] && this._rdKitMols[i]?.is_qmol)
+          break;
+        case Fingerprint.Morgan:
+          for (let i = 0; i < this._rdKitMols.length; ++i) {
+            try {
+              if (!this._rdKitMols[i] || this._rdKitMols[i] && this._rdKitMols[i]?.is_qmol)
+                fps.push(null);
```

Comment:
Why would we push(null) instead of allocating an array of the right size upfront, and then just setting i-th element?

Reply:
Corrected